kratos/geometries/quadrilateral_2d_4.h

```diff
@@ -428,6 +428,32 @@ template<class TPointType> class Quadrilateral2D4
         //return fabs(DeterminantOfJacobian(PointType())) * 0.5;
     }
 
+    /// detect if quadrilateral and box are intersected
+    bool HasIntersection( const Point<3, double>& rLowPoint, const Point<3, double>& rHighPoint ) override {
+        
+        const BaseType& geom_1 = *this;
+
+        Point<3, double> boxcenter;
+        Point<3, double> boxhalfsize;
+
+        boxcenter[0]   = 0.50 * (rLowPoint[0] + rHighPoint[0]);
+        boxcenter[1]   = 0.50 * (rLowPoint[1] + rHighPoint[1]);
+        boxcenter[2]   = 0.00;
+
+        boxhalfsize[0] = 0.50 * (rHighPoint[0] - rLowPoint[0]);
+        boxhalfsize[1] = 0.50 * (rHighPoint[1] - rLowPoint[1]);
+        boxhalfsize[2] = 0.00;
+
+        std::size_t size = geom_1.size();
+        std::vector<Point<3, double> > quadverts;
+        quadverts.resize(size);
+        for(unsigned int i = 0; i< size; i++ )
+            quadverts[i] =  geom_1.GetPoint(i);
+
+        bool result = false;
+        result      = QuadBoxOverlap(boxcenter, boxhalfsize, quadverts);
+        return result;
```

Comment:
This can be simplified:
~~~cpp
return QuadBoxOverlap(boxcenter, boxhalfsize, quadverts);
~~~
or even:
~~~cpp
const bool result = QuadBoxOverlap(boxcenter, boxhalfsize, quadverts);

return result;
~~~